datadog_checks_dev/datadog_checks/dev/plugin/pytest.py

```diff
@@ -136,17 +138,28 @@ def dd_environment_runner(request):
     else:  # no cov
         # Exit testing and pass data back up to command
         if E2E_RESULT_FILE in os.environ:
-            # Standard `pytest.exit` requires a reason but we want an empty string for minimal output
-            from _pytest.outcomes import Exit
-
             with open(os.environ[E2E_RESULT_FILE], 'w', encoding='utf-8') as f:
                 f.write(json.dumps(data))
 
-            Exit('', 0)
+            # Rather than exiting we skip every test to avoid the following output:
+            # !!!!!!!!!! _pytest.outcomes.Exit: !!!!!!!!!!
+            pytest.skip()
         else:
             pytest.exit(message_template.format(serialize_data(data)))
 
 
+# Manipulate the output if we are spinning up or down an environment
+if set_up_env() or tear_down_env():
+
+    def pytest_report_teststatus(report, config):
+        """
+        https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_report_teststatus
+        """
+        # Skipping every test displays an `s` for each even when using
+        # the minimum verbosity so we force zero output
+        return 'skipped', '', ''
+
+
```

Comment:
I'm not that familiar neither with the e2e machinery for our plugin nor with pytest plugins in general, but I wonder if using the [`pytest_collection`](https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_collection) hook instead would work, e.g. something like

```suggestion
# Avoid regular pytest test collection if we are spinning up or down an environment
if set_up_env() or tear_down_env():

    def pytest_collection(report, config):
        """
        https://docs.pytest.org/en/stable/reference/reference.html#pytest.hookspec.pytest_collection
        """
        # Any non-None value stops test collection
        return True


```

Or any other collection-related hook that would remove tests rather than skip them, avoiding the extra step of having to suppress any extraneous output.

Reply:
After a closer look, I think I see why that's probably not possible (because we can only run the `dd_environment_runner` fixture _after_ collection).

Reply:
I can confirm that if collection is disabled then the fixture does not run